BreastCancer/TasksAndSteps/APHExerciseCheckinTask.m

```diff
@@ -0,0 +1,560 @@
+            ORKQuestionStep*            step                = [ORKQuestionStep questionStepWithIdentifier:kUserUncertainID
+                                                                                                    title:NSLocalizedString(kUncertainTitle, nil)
+                                                                                                   answer:pickerAnswerFormat];
+        
+            step.text       = NSLocalizedString(kUncertainDetail, nil);
+            step.optional   = NO;
+            
+            [steps addObject:step];
+        }
+        {
+            ORKInstructionStep* step = [[ORKInstructionStep alloc] initWithIdentifier:kReasonsStepID];
+            
+            step.title      = NSLocalizedString(@"No reason", nil);
+            step.detailText = NSLocalizedString(@"None at all", nil);
+            
+            [steps addObject:step];
+        }
+        {
+            ORKInstructionStep* step = [[ORKInstructionStep alloc] initWithIdentifier:kReasonsTwoStepID];
+            
+            step.title      = NSLocalizedString(@"No reason", nil);
+            step.detailText = NSLocalizedString(@"None at all", nil);
+            
+            [steps addObject:step];
+        }
+    }
+    
+    self = [super initWithIdentifier:kExerciseCheckInID steps:steps];
+    
+    return self;
+}
+
+- (nullable ORKStep *)stepAfterStep:(nullable ORKStep *)step withResult:(ORKTaskResult *)result
+{
+    self.lastStep = step;
+    
+    NSInteger stepIndex = 0;
+    
+    NSInteger(^DetermineStepDepth)(ORKStep*, ORKTaskResult*) = ^NSInteger(ORKStep* step, ORKTaskResult* __unused result)
+    {
+        NSInteger depth = 0;
+        
+        if ([step.identifier isEqualToString:kInstructionStepID])
+        {
+            depth = 1;
+        }
+        else if ([step.identifier isEqualToString:kUserProgressStepID])
+        {
+            depth = 2;
+        }
+        else if ([step.identifier isEqualToString:kUserSucceededStepID] || [step.identifier isEqualToString:kUserFailedStepID] || [step.identifier isEqualToString:kUserUncertainID])
+        {
+            depth = 3;
+        }
+        else if ([step.identifier isEqualToString:kUserInputStepID])
+        {
+            depth = 4;
+        }
+        
+        return depth;
+    };
+    
+    NSInteger(^DepthTwoResult)(ORKStep*, ORKTaskResult*) = ^NSInteger(ORKStep* step, ORKTaskResult* result)
+    {
+        NSInteger stepIndex = kNegativeStepIndex;
+        
+        ORKStepResult* stepResult = (ORKStepResult*)[result resultForIdentifier:step.identifier];
+        
+        if (stepResult)
+        {
+            ORKChoiceQuestionResult* choiceResult = (ORKChoiceQuestionResult*)[stepResult firstResult];
+            
+            if (choiceResult.choiceAnswers)
+            {
+                NSArray* answers        = choiceResult.choiceAnswers;
```

Comment:
I'll create a Github issue to track this but using a numeric value would affect how data is collected and, therefore, impact the researchers. Resolution of this would require working with the researchers to ensure no data lost or issue.